src/cpp_source/ut_gtest.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2023 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+#include <ut.h>
+#include <ut_internal.h>
+#include <ut_test_runner.h>
+
+void UT_set_results_output_filename(const char* szFilenameRoot)
+{
+    std::string filepath = szFilenameRoot;
+
+    // Find the position of the last dot
+    size_t lastDot = filepath.find_last_of('.');
+
+    // Strip the extension if found
+    if (lastDot != std::string::npos)
+    {
+        filepath = filepath.substr(0, lastDot);
+    }
+
+    // Set the output format and path programmatically
+    ::testing::FLAGS_gtest_output = std::string("xml:") + filepath + "-report.xml";
+}
+
+void UT_set_test_mode(TestMode_t  mode)
+{
+    return;
+}
+
+void UT_Manage_Suite_Activation(int groupID, bool enable_disable)
+{
+    return;
+}
+
+void UT_toggle_suite_activation_based_on_groupID(UT_groupID_t groupId, bool enable_disable)
+{
+    return;
+}
+
+UT_status_t startup_system( void )
+{
+    return UT_STATUS_OK;
```

Comment:
how is this installing gtest and making it work?

Reply:
startup_system() is a pass through function. Doesn't do anything. It was needed as its being called inside ut_main.c, UT_init().